docs/source/remote-data-services.rst

```diff
@@ -87,36 +88,62 @@ The Hadoop File System (HDFS) is a widely deployed, distributed, data-local file
 system written in Java. This file system backs many clusters running Hadoop and
 Spark.
 
-Within dask, HDFS is only available when the module ``distributed.hdfs`` is
-explicitly imported, since the usage of HDFS usually only makes sense in a cluster
-setting. The distributed scheduler will prefer to allocate tasks
-which read from HDFS  to machines which have local copies of the
-blocks required for their work, where possible.
+HDFS support can be provided by either hdfs3_ or pyarrow_, defaulting to the
+first library installed in that order. To explicitly set which library to use,
+set ``hdfs_driver`` using ``dask.set_options``:
+
+.. code-block:: python
+
+   # Use first available option in [hdfs3, pyarrow]
+    dd.read_csv('hdfs:///path/to/*.csv')
+
+   # Use hdfs3 for HDFS I/O
+   with dask.set_options(hdfs_driver='hdfs3'):
+       dd.read_csv('hdfs:///path/to/*.csv')
+
+   # Use pyarrow for HDFS I/O
+   with dask.set_options(hdfs_driver='pyarrow'):
+       dd.read_csv('hdfs:///path/to/*.csv')
+
+   # Set pyarrow as the global hdfs driver
+   dask.set_options(hdfs_driver='pyarrow')
 
-By default, hdfs3 attempts to read the default server and port from local
-Hadoop configuration files on each node, so it may be that no configuration is
-required. However, the server, port and user can be passed as part of the
-url: ``hdfs://user:pass@server:port/path/to/data``.
 
-The following parameters may be passed to hdfs3 using ``storage_options``:
+By default, both libraries attempt to read the default server and port from
+local Hadoop configuration files on each node, so it may be that no
+configuration is required. However, the server, port and user can be passed as
+part of the url: ``hdfs://user:pass@server:port/path/to/data``.
 
-    - host, port, user: basic authentication
-    - ticket_cache, token: kerberos authentication
-    - pars: dictionary of further parameters (e.g., for `high availability`_)
+Extra Configuration for HDFS3
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following additional options may be passed to the ``hdfs3`` driver via
+``storage_options``:
+
+    - ``host``, ``port``, ``user``: basic authentication
+    - ``ticket_cache``, ``token``: kerberos authentication
+    - ``pars``: dictionary of further parameters (e.g., for `high availability`_)
+
+The ``hdfs3`` driver also relies on a few environment variables. For
```

Comment:
The ``hdfs3`` driver configuration can also be affected by a few environment variables.